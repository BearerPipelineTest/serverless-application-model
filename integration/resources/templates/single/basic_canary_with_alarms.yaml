Resources:
  SyntheticsCanaryAlarm:
    Type: AWS::Serverless::Canary
    Properties:
      Handler: pageLoadBlueprint.handler
      InlineCode: |
        def handler(event, context):
          print("Hello, world!")
      Runtime: syn-python-selenium-1.0
      StartCanaryAfterCreation: true
      Schedule:
        Expression: rate(0 minute)
      CanaryMetricAlarms:
        - AlarmWithEverythingDefined:
            MetricName: SuccessPercent
            Threshold: 100
            ComparisonOperator: GreaterThanThreshold
            Period: 11160
        - SuccessPercentAlarm:
            MetricName: SuccessPercent
        - DurationAlarm:
            MetricName: Duration
        - FailedAlarm:
            MetricName: Failed
