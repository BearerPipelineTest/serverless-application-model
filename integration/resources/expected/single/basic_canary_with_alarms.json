[
  { "LogicalResourceId":"SyntheticsCanaryAlarm", "ResourceType":"AWS::Synthetics::Canary" },
  { "LogicalResourceId":"SyntheticsCanaryAlarmRole", "ResourceType":"AWS::IAM::Role" },
  { "LogicalResourceId":"SyntheticsCanaryAlarmArtifactBucket", "ResourceType":"AWS::S3::Bucket" },
  { "LogicalResourceId":"AlarmWithEverythingDefined", "ResourceType":"AWS::CloudWatch::Alarm" },
  { "LogicalResourceId":"SuccessPercentAlarm", "ResourceType":"AWS::CloudWatch::Alarm" },
  { "LogicalResourceId":"DurationAlarm", "ResourceType":"AWS::CloudWatch::Alarm" },
  { "LogicalResourceId":"FailedAlarm", "ResourceType":"AWS::CloudWatch::Alarm" }
]
